dangerzone/gui/updater.py

```diff
@@ -0,0 +1,260 @@
+            return True
+        else:
+            return False
+
+    def get_latest_info(self) -> UpdateReport:
+        """Get the latest release info from GitHub.
+
+        Also, render the changelog from Markdown format to HTML, so that we can show it
+        to the users.
+        """
+        try:
+            # TODO: Set timeout.
+            info = requests.get(self.GH_RELEASE_URL).json()
+        except Exception as e:
+            # TODO:: Handle errors.
+            raise
+
+        version = info["tag_name"].lstrip("v")
+        changelog = markdown.markdown(info["body"])
+
+        return UpdateReport(version=version, changelog=changelog)
+
+    # XXX: This happens in parallel with other tasks. DO NOT alter global state!
+    def _check_for_updates(self) -> UpdateReport:
+        """Check for updates locally and remotely.
+
+        Check for updates in two places:
+
+        1. In our settings, in case we have cached the latest version/changelog from a
+           previous run.
+        2. In GitHub, by hitting the latest releases API.
+        """
+        log.debug(f"Checking for Dangerzone updates")
+        latest_version = self.dangerzone.settings.get("updater_latest_version")
+        if version.parse(get_version()) < version.parse(latest_version):
+            log.debug(f"Determined that there is an update due to cached results")
+            return UpdateReport(
+                version=latest_version,
+                changelog=self.dangerzone.settings.get("updater_latest_changelog"),
+            )
+
+        # If the previous check happened before the cooldown period expires, do not
+        # check again. Else, bump the last check timestamp, before making the actual
+        # check. This is to ensure that even failed update checks respect the cooldown
+        # period.
+        if self._should_postpone_update_check():
+            return UpdateReport()
+        else:
+            self.dangerzone.settings.set(
+                "updater_last_check", self._get_now_timestamp(), autosave=True
+            )
+
+        log.debug(f"Checking the latest GitHub release")
+        report = self.get_latest_info()
+        log.debug(f"Latest version in GitHub is {report.version}")
+        if report.version and self.can_update(latest_version, report.version):
+            log.debug(
+                f"Determined that there is an update due to a new GitHub version:"
+                f" {latest_version} < {report.version}"
+            )
+            return report
+
+        log.debug(f"No need to update")
+        return UpdateReport()
+
+    ##################
+    # Logic for running update checks asynchronously
+
+    def check_for_updates(self) -> UpdateReport:
+        """Check for updates and return a report with the findings:
+
+        There are three scenarios when we check for updates, and each scenario returns a
+        slightly different answer:
+
+        1. No new updates: Return an empty dictionary.
```

Comment:
This one needs to be updated now that this is not a dict

Reply:
Also the lines after

Reply:
Yikes, good catch. Fixed in 4e01321b51f7a3948d1d4b1db32ad4fe9fd165b9.